### PR TITLE
Add flag and logic to removal form to make otp for device removal optional

### PR DESCRIPTION
**Pull request by jkaeske**
I tried to implement the feature specified in #169.
I also wrote a test for the device removal with and without required otp.

Would be nice to get a review on that :)

### CODE REVIEW

File: tests/test_allauth_2fa.py [hunk 1]
```diff
@@ -253,6 +255,28 @@ def test_2fa_removal(client, john_with_totp, token_state):
...
+        expected_redirect_url=settings.LOGIN_REDIRECT_URL,
+    )
+    assert user.totpdevice_set.exists()
+
+    # Navigate to 2FA removal view
+    client.get(reverse("two-factor-remove"))
+    # ... and POST to confirm
+    client.post(reverse("two-factor-remove"), {})
+
+    # The only case when the TOTP device should be removed is when the token is correct.
```
Comment (akx) [inline tests/test_allauth_2fa.py]: This copy-pasted comment is wrong for this test. You could combine this with the above test using `pytest.mark.parametrize` anyhow –

```python
@pytest.mark.parametrize("require_token", [False, True])
def test_2fa_removal(client, john_with_totp, token_state, require_token, monkeypatch):
    monkeypatch.setattr(app_settings, "REQUIRE_OTP_ON_DEVICE_REMOVAL", require_token)
```
and then at the very end verify that if `require_token` is false, the token is removed no matter the user input?

Comment (jkaeske) [reply tests/test_allauth_2fa.py]: Thank you, that makes sense!
I will adjust it later

**Comment by jkaeske** [comment]
That's done @akx

**Review by akx — APPROVED** [review]
LGTM, thanks!

**Comment by valberg** [comment]
Great work @jkaeske!

**Comment by valberg** [comment]
@jkaeske Can you do an update to CHANGELOG.md as well?

**Comment by jkaeske** [comment]
I did that and created a pull request

**Pull request merged**